src/pages/Course/Course.hooks.tsx

```diff
@@ -0,0 +1,142 @@
+ * Hooks for the Course page.
+ * @returns An object with the functions calculateTopicProgress and BorderLinearProgress.
+ * @remarks
+ * Uses the {@link calculatePercent} function to calculate the progress of a topic.
+ * Uses the {@link colorByPercent} function to calculate the color of the progress bar.
+ * Uses the {@link calculateTopicProgress} function to calculate the progress of a topic.
+ * Uses the {@link BorderLinearProgress} styled component to style the progress bar.
+ * @example progress variation
+ * circular Progress
+ * ```
+ *
+ *    Another possibility to show the progress of the course in a circle
+ *
+ *    import CircularProgress, {
+ *    CircularProgressProps,
+ *    } from '@mui/material/CircularProgress'
+ *    function CircularProgressWithLabel(
+ *    props: CircularProgressProps & { value: number } & { text: string},
+ *    ) {
+ *    return (
+ *    <Box sx={{ position: 'relative', display: 'inline-flex' }}>
+ *    <CircularProgress variant="determinate" {...props} />
+ *    <Box
+ *    sx={{
+ *    top: 0,
+ *    left: 0,
+ *    bottom: 0,
+ *    right: 0,
+ *    position: 'absolute',
+ *    display: 'flex',
+ *    alignItems: 'center',
+ *    justifyContent: 'center',
+ *    }}
+ *    >
+ *    <Typography
+ *    variant="caption"
+ *    color="text.primary"
+ *    >{`${(props.text)}`}</Typography>
+ *    </Box>
+ *    </Box>
+ *    )
+ *    }
+ * ```
+ * @category Hooks
+ */
+export const useCourse = (): CourseHookReturn => {
+  //Linear Progress Bar with label
+  const LinearProgressWithLabel = (
+    props: LinearProgressProps & { value: number } & { text: string } & {
+      textposition: { xs: string; sm: string; md: string; lg: string; xl: string }
+    }
+  ) => {
+    return (
+      <div>
+        <Tooltip title={'Completed learning elements'}>
+          <Typography sx={{ ml: props.textposition, mr: '0.5rem' }} variant="body1" color="text.secondary">
+            {'Learning progress: ' + props.text}
+          </Typography>
+        </Tooltip>
+        <LinearProgress
+          variant="determinate"
+          {...props}
+          sx={{ ml: { xs: '16rem', sm: '0rem', md: '-7rem', lg: '-7rem', xl: '-11rem' } }}
+        />
+      </div>
+    )
+  }
+
+  //styled Linear Progress Bar
+  const BorderLinearProgress = styled(LinearProgressWithLabel)(({ theme }) => ({
+    height: 7,
+    borderRadius: 5,
+    [`&.${linearProgressClasses.colorPrimary}`]: {
+      backgroundColor: theme.palette.grey[800]
+    },
+    [`& .${linearProgressClasses.bar}`]: {
+      borderRadius: 3
+    }
+  }))
+
+  //calculate the progress of a topic and return a progress bar with variable text, color and value
+  //learningElementProgressTopics is an array of arrays with the progress of each topic
+  //first element of the array is the number of solved learning elements,
+  //second element is the number of total learning elements
+  const calculateTopicProgress = (learningElementProgressTopics: number[][], index: number) => {
```

Comment:
Das returned ein JSX.Element und sollte daher groß geschrieben werden. Allerdings könnte es eh sinnvoll sein Komponenten in eine extra File zu packen und nicht innerhalb der Hook anzulegen.

Reply:
neue Komponente heißt nun LinearProgressWithLabel